components/properties/src/names.rs

```diff
@@ -257,18 +257,55 @@ impl<T: TrieValue> PropertyParserBorrowed<'static, T> {
 
 /// Avoid monomorphizing multiple copies of this function
 fn get_strict_u16(payload: &PropertyValueNameToEnumMapV1<'_>, name: &str) -> Option<u16> {
-    // NormalizedPropertyName has no invariants so this should be free, but
-    // avoid introducing a panic regardless
-    let name = NormalizedPropertyNameStr::parse_byte_slice(name.as_bytes()).ok()?;
-    payload.map.get_copied(name)
+    payload.map.get(name).and_then(|i| i.try_into().ok())
 }
 
 /// Avoid monomorphizing multiple copies of this function
 fn get_loose_u16(payload: &PropertyValueNameToEnumMapV1<'_>, name: &str) -> Option<u16> {
-    // NormalizedPropertyName has no invariants so this should be free, but
-    // avoid introducing a panic regardless
-    let name = NormalizedPropertyNameStr::parse_byte_slice(name.as_bytes()).ok()?;
-    payload.map.get_copied_by(|p| p.cmp_loose(name))
+    fn recurse(mut cursor: ZeroTrieSimpleAsciiCursor, name: &[u8]) -> Option<usize> {
+        if cursor.is_empty() {
+            return None;
+        }
+
+        // Skip whitespace, underscore, hyphen in trie.
+        for skip in [b'\t', b'\n', b'\x0C', b'\r', b' ', 0x0B, b'_', b'-'] {
```

Comment:
Question: do these characters like `\t` and `\n` actually occur in the trie? Seems unlikely?

Reply:
Not in today's compiled data, but according to the spec this is what we have to do.

Reply:
Citation in the spec?

You say "fairly cheap", but this is sort-of a hot path (for example, regex and unicode set parsing), and every one of these requires a function call, and function calls are not free. My guess is that these extra function calls together make the function about 2x as slow. I could be wrong.

I also find it _incredibly_ unlikely that the UCD would add a canonical property name containing characters like `\t` and `\n`. I understand skipping those in the user's string, but not in the trie string.